variables.tf

```diff
@@ -1,4 +1,75 @@
-variable "example" {
-  description = "Example variable"
-  default     = "hello world"
+variable "create_sns_topic" {
+  description = <<-DOC
+  Flag to indicate whether an SNS topic should be created for notifications
+  If you want to send findings to a new SNS topic, set this to true and provide a valid configuration for subscribers
+  DOC
+
+  type    = bool
+  default = false
 }
+
+variable "subscribers" {
+  type = map(object({
+    protocol               = string
+    endpoint               = string
+    endpoint_auto_confirms = bool
+  }))
+  description = <<-DOC
+  A map of subscription configurations for SNS topics
+    
+  For more information, see:
+  https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sns_topic_subscription#argument-reference
+ 
+  protocol:         
+    The protocol to use. The possible values for this are: sqs, sms, lambda, application. (http or https are partially 
+    supported, see link) (email is an option but is unsupported in terraform, see link).
+  endpoint:         
+    The endpoint to send data to, the contents will vary with the protocol. (see link for more information)
+  endpoint_auto_confirms:
+    Boolean indicating whether the end point is capable of auto confirming subscription e.g., PagerDuty. Default is 
+    false
+  DOC
+  default     = {}
+}
+
+variable "findings_notification_arn" {
+  description = <<-DOC
+  The ARN for an SNS topic to send findings notifications to. This is only used if create_sns_topic is false.
```

Comment:
Personal preference would be to indent one more level. The whitespace will still be stripped.

Reply:
7bf22db922b95693c5372a0f2d0e43d1ead9539a